### PR TITLE
[MU4] Convert some enums to enum classes

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Fixes a bunch of MSVC IntelliSense warnings

* `INotationParts::InsertMode`
* `LayoutBreak::Type`
* `AbstractInspectorModel::InspectorSectionType`
* `AbstractInspectorProxyModel::InspectorModelType`
* `SystemDivider::Type`
* `Ms:MsError`
* `mu::midi::SynthesizerState::ValID`

ToDo:

* `Ms::EventType`, being deprecated this might not be worth the effort though? For sure it'd be getting pretty ugly...

In the passing initialize some member variables that MSVC  was complaing about

Not changing, as these are in third party code:

* `deto::async::Channel::Type`
* `haw::Logger::Level`
* Lots from Qt

### CODE REVIEW

File: thirdparty/deto_async/async/changednotify.h [hunk 1]
```diff
@@ -163,20 +163,20 @@ class ChangedNotify
         {
             CallType type = static_cast<CallType>(_type);
             switch (type) {
-            case Undefined: {} break;
-            case Changed:     {
+            case CallType::Undefined: {} break;
```
Comment (Marr11317) [inline thirdparty/deto_async/async/changednotify.h:166]: since when do we put braces in `case` statements?

Comment (Marr11317) [reply thirdparty/deto_async/async/changednotify.h:166]: yes, sorry the question wasn't to you in particular. Anyway, it does no harm.

**Comment by Jojo-Schmitz** [comment]
Strange, shows a bunch  of build errors I did not see locally...

File: thirdparty/deto_async/async/changednotify.h (hunk 1 above)
Comment (Jojo-Schmitz) [reply thirdparty/deto_async/async/changednotify.h:165]: Just noticed: this is third party code!

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**